File: examples/basic.py
"""Minimal LangGraph (Python) example: ask a human inside a node.

Prereqs: pip install pushary-langgraph
Run:     PUSHARY_API_KEY=... python examples/basic.py
"""
from pushary_langgraph import ask_human, connect

USER_ID = "user_123"


def main() -> None:
    # 1) One time per end-user: connect their phone.
    link = connect(USER_ID)
    print("Ask the user to open:", link)

    # 2) Ask a real human, block, fail-closed.
    decision = ask_human("Approve a $40 refund?", external_id=USER_ID, node="approval")
    print("approved:", decision["approved"])


if __name__ == "__main__":
    main()

File: examples/basic.ts
/**
 * Minimal LangGraph example: give a ReAct agent a real human to ask.
 *
 * Prereqs: npm i @pushary/langgraph @langchain/langgraph @langchain/core @langchain/openai
 * Run:     PUSHARY_API_KEY=... OPENAI_API_KEY=... npx tsx examples/basic.ts
 */
import { ChatOpenAI } from '@langchain/openai'
import { createReactAgent } from '@langchain/langgraph/prebuilt'
import { connect, createAskHumanTool } from '@pushary/langgraph'

const config = { apiKey: process.env.PUSHARY_API_KEY! }
const userId = 'user_123'

async function main() {
  // 1) One time per end-user: connect their phone.
  const { universalLink } = await connect(config, userId)
  console.log('Ask the user to open:', universalLink)

  // 2) Give the agent a tool that blocks on a real human, fail-closed.
  const askHuman = createAskHumanTool(config, { externalId: userId })
  const agent = createReactAgent({
    llm: new ChatOpenAI({ model: 'gpt-4o' }),
    tools: [askHuman],
  })

  const res = await agent.invoke({
    messages: [{ role: 'user', content: 'Issue a $40 refund, but only if a human approves.' }],
  })
  console.log(res.messages.at(-1)?.content)
}

main().catch((err) => {
  console.error(err)
  process.exit(1)
})
